### ISSUE TITLE
TFRS Retrieval Model with softmax for converting score of each items into probabilities 

### ISSUE THREAD

**Issue by drtinumohan** [tensorflow/recommenders#522]
We are using TFRS as the recommendation solution for our usecase. Instead of returning the score associated with each item we passed the scores into the softmax and returning probability score of each item. Is there any another way to figure out the relative probability of items? 
![TFRS_with_softmax](https://user-images.githubusercontent.com/48860927/179494595-a112e55c-0196-4342-890a-df607f633626.png)

**Comment by patrickorlando** [comment]
Its not very common to require the probabilities, but if you do, you need to compute the scores for all candidates, which will only work in the brute force retrieval. 

If all you care about is the ratio of probabilities, then you can compute this on a subset of your scores. 

![image](https://user-images.githubusercontent.com/17157991/180770723-92b2895d-58c8-4d07-9669-322059665c5d.png)
